nexus/reconfigurator/planning/src/planner.rs

```diff
@@ -148,6 +161,22 @@ impl<'a> Planner<'a> {
                 (SledPolicy::Expunged, SledState::Active) => (),
             }
 
+            // Check that the sled isn't already decommissioned in the parent blueprint, if it exists.
+            if let Some(sled_state) =
+                self.blueprint.parent_blueprint().sled_state.get(&sled_id)
+            {
+                if *sled_state == SledState::Decommissioned {
+                    continue;
+                }
+            }
+
+            // Decommission any disks that need it
+            //
+            // This call must live here so that we don't try to decommission disks
+            // on an already decommissioned sled. That violates editor invariants,
+            // as the planner should not be trying to edit sleds that have been pulled.
```

Comment:
Nit - I'd be specific here, since once we have trust quorum decommissioning won't require physical removal.

```suggestion
            // as the planner should not be trying to edit sleds that have been decommissioned.
```

Reply:
Good eye. Done in de4a0805309651eea73e6c8bab1e256afbaf992d.